tile_fetch.py

```diff
@@ -138,12 +166,35 @@ async def load_tiles(info, z=-1, outfile=None, quality=90):
         img.paste(tile_img, (x * info.tile_width, y * info.tile_height))
 
     print("Downloaded all tiles. Saving...")
-    final_image_filename = outfile or (info.image_name + '.jpg')
-    img.save(final_image_filename, quality=quality, subsampling=0)
+    
+  ## Try to extract author name ("Creator"/"Painter") and date ("Date Created"/"Date") from metadata
+    author = "0"
+    date = ""
+    for key, value in info.metadata.items():
+        if key.lower() == "xmp.xmp.creator" or key.lower() == "xmp.xmp.painter":
+            # Avoiding non-ASCII characters in the painter/creator name
+            author = unidecode.unidecode(value)
+        elif key.lower() == "xmp.xmp.date" or key.lower() == "xmp.xmp.datecreated":
+            # Avoiding "/" in the date (year), especially when multiple dates are given
+            date = value.replace('/','-')
+            
+    # Taking out the author's name from the image name - authors name is appended later
+    modified_image_name = info.image_name[0:len(info.image_name)-len(author)-1]
+    
+    final_image_filename = (author + ' - ' + date + ' - ' + modified_image_name + ' - ' +info.image_id + '.jpg')
+    img.save(final_image_filename, quality=quality, subsampling=0, optimize=True)
+    xmp_file_obj = TaggedImage(final_image_filename)
+
+    # writes key:value one at a time, which is heavier on writes,
+    # but far more robust.
```

Comment:
In what situation can tag writes fail ? Can't it be known ahead of time ?

Reply:
My friend added this catch-code - but I believe some foreign characters (Japanese, perhaps? Something unsupported on the host system) can fail. We came across one such case, but I forget what the details were behind it.

Reply:
Can you try to find the cases that make this fail ?

Reply:
As pere [here](https://github.com/LeoHsiao1/pyexiv2) the limit seems to be 2gb. Maybe the image before being saved is larger than that in memory?

Reply:
> Can you try to find the cases that make this fail ?
> 
> Edit: ok, I just saw your comment below, thanks

In addition, I think [this link](https://artsandculture.google.com/asset/winter-landscape-with-bird-trap/mgG0Fs1WLFP5rg) prompted my friend to write the key:values line by line. I've attached the error message I originally sent him,
![python_2](https://user-images.githubusercontent.com/34801180/87449360-19329600-c5cb-11ea-9462-f39f749c3c37.png)

Reply:
Looking at it further I am seeing a slowdown in the last piece of the code... am wondering if writing the XMP tags one by one is time consuming. Will definitely think about this...